leader/src/cli.rs

```diff
@@ -33,7 +33,14 @@ pub(crate) enum Command {
         rpc_url: String,
         /// The block number for which to generate a proof.
         #[arg(short, long)]
-        block_number: u64,
+        block_number: Option<u64>,
+        /// The starting block number of the range from which to generate a
+        /// proof.
+        #[arg(long)]
+        from: Option<u64>,
+        /// The ending block number of the range to which to generate a proof.
+        #[arg(long)]
+        to: Option<u64>,
```

Comment:
Since `block_number` & `from`/`to` are mutually exclusive, wdyt about parsing both to a single prog arg? I actually wrote logic to do exactly this:

```rust
#[derive(Clone, Debug)]
pub(super) enum BlockNumbers {
    Single(u64),
    RangeInclusive(RangeInclusive<u64>),
}

impl FromStr for BlockNumbers {
    type Err = anyhow::Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::from_str_intern(s)
            .with_context(|| {
                format!(
                    "Expected a single value or a range, but instead got \"{}\".",
                    s
                )
            })
            .map_err(|e| anyhow!("{e:#}"))
    }
}

impl BlockNumbers {
    fn from_str_intern(s: &str) -> anyhow::Result<Self> {
        // Did we get passed a single value?
        if let Ok(v) = s.parse::<u64>() {
            return Ok(Self::Single(v));
        }

        // Check if it's a range.
        let mut range_vals = s.split("..=");

        let start = Self::next_and_try_parse(&mut range_vals)?;
        let end = Self::next_and_try_parse(&mut range_vals)?;

        if range_vals.count() > 0 {
            return Err(anyhow!(
                "Parsed a range but there were unexpected characters afterwards!"
            ));
        }

        Ok(Self::RangeInclusive(start..=end))
    }

    fn next_and_try_parse(range_vals: &mut Split<&str>) -> anyhow::Result<u64> {
        let unparsed_val = range_vals
            .next()
            .with_context(|| "Parsing a value as a `RangeInclusive`")?;
        let res = unparsed_val
            .parse()
            .with_context(|| format!("Parsing the range val \"{unparsed_val}\" into a usize"))?;

        Ok(res)
    }
}

pub(crate) enum Command {
...
        /// The block numbers for which to generate proofs.
        #[arg(short, long)]
        block_numbers: BlockNumbers,
...
}
```

Reply:
I had it initially exactly the way you're suggesting, to take `--block-numbers 14..20`, later I changed my mind, not sure why I changed my mind but yes, I can return it back to use a single argument

Reply:
Yeah you're call! Just my personal preference.